### PR TITLE
Use repo's default branch if no branch specified

### LINKED ISSUE #74: Use repo default branch instead of hard-coded "master"
https://github.com/vsoch/pull-request-action/blob/1cb1956a68e7d8b6bf693d41299467bb0f1b6a7a/pull-request.py#L347

It probably makes sense to use the relevant API call to fetch the default branch of a repo instead of hard-coding in "master" here.  Github [made the change](https://github.blog/changelog/2020-07-22-github-actions-better-support-for-alternative-default-branch-names/) last year, and the way to get the value in a shell is
```shell
curl -L -H 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' \
            https://api.github.com/repos/${owner}/${repo} \
            | jq .default_branch
```

That should be pretty straightforward to translate into Python.  I'll open a PR when I find a few spare minutes if no one else does it first.

### PR DESCRIPTION
**Pull request by dannysauer**
Add find_default_branch() method to fetch default branch for repo.  Method should only be called if env var isn't specified, I think, so there shouldn't be a performance impact.

Closes #74

---
I haven't tested as a complete action yet, but here's validation of the important code on the command line:
```python
09:37 $ python3
Python 3.6.12 (default, Dec 02 2020, 09:44:23) [GCC] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
>>> REPO_URL="https://api.github.com/repos/Kong/pull-request-action"
>>> response = requests.get(REPO_URL)
>>> response.status_code
200
>>> response.json()["default_branch"]
'master'
>>> REPO_URL="https://api.github.com/repos/Kong/docker-pulp"
>>> response = requests.get(REPO_URL)
>>> response.status_code
200
>>> response.json()["default_branch"]
'main'
```

### CODE REVIEW

File: pull-request.py [hunk 1]
```diff
@@ -234,6 +234,20 @@ def find_pull_request(listing, source):
...
+    """Find default branch for a repo (only called if branch not provided)
+    """
+    response = requests.get(REPO_URL)
+
+    # Case 1: 404 might need a token
+    if response.status_code == 404:
+        response = requests.get(REPO_URL, headers=HEADERS)
+    if response.status_code != 200:
+        abort_if_fail(response, "Unable to retrieve default branch")
+
```
Comment (vsoch) [inline pull-request.py:247]: Could  we put one more line here (before returning anything) that says something like "Found default branch X"? That way you can easily see in the action that the function was being used (and what it returned).

Comment (dannysauer) [reply pull-request.py:247]: Sure thing.  I was originally planning to do that, and then completely forgot about it when I wrote the actual code. :D

I used an f-string, but then as I was hitting return on the `git push`, I noticed the rest of the code uses the printf-style string replacements.  Want me to change it to match?

Comment (vsoch) [reply pull-request.py:247]: I would be conservative and use print style string replacements.

Comment (dannysauer) [reply pull-request.py:247]: Ok - printf'd Line 249

**Comment by vsoch** [comment]
Looks good @dannysauer ! https://github.com/vsoch/pull-request-action-example/runs/2877041518?check_suite_focus=true. Let's merge and release.

**Review by vsoch — APPROVED** [review]

**Pull request merged**